### PR TITLE
Updated kubernetes cluster version to 1.28 and update GPU operator to newest

### PR DESCRIPTION
**Pull request by MaggieXJZhang**
This MR updates the following to match with CNS release 11.0

1. kubernetes version to 1.28. (Note: on GKE, we're setting a min_version with a new variable so that we don't rely on GKE's default value for regular channel, which is 1.27 right now, but this is only a min_version and does not guarantee exact versioning. See more on [min_master_version](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#min_master_version)
2. GPU operator version 23.9.1 and driver version to 535.129.03
3. NVAIE GPU operator version to 23.9.0 and driver version to 535.129.03
4. Update the copyright header to 2024 for all revised files
5. Renegenerate the markdown docs and tfvar defaults
6. Updated contributing.md 

## AKS 

```
~ kubectl get nodes
NAME                                   STATUS   ROLES   AGE     VERSION
aks-holoscancpu-27424565-vmss000000    Ready    agent   15m     v1.28.3
aks-holoscangpu1-21964465-vmss000000   Ready    agent   9m54s   v1.28.3
aks-holoscangpu1-21964465-vmss000001   Ready    agent   10m     v1.28.3

~ helm ls -n gpu-operator
NAME        	NAMESPACE   	REVISION	UPDATED                                	STATUS  	CHART               	APP VERSION
gpu-operator	gpu-operator	1       	2024-01-10 12:05:23.428413728 -0500 EST	deployed	gpu-operator-v23.9.1	v23.9.1 

~ kubectl exec -it nvidia-device-plugin-daemonset-mjv7f -n gpu-operator -- nvidia-smi
Defaulted container "nvidia-device-plugin" out of: nvidia-device-plugin, toolkit-validation (init)
Wed Jan 10 17:12:01 2024       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.85.12    Driver Version: 525.85.12    CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-PCIE...  On   | 00000001:00:00.0 Off |                  Off |
| N/A   30C    P0    24W / 250W |      0MiB / 16384MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

```

## EKS

```
☁  eks [magzhang/update-to-k8s-1.28] ⚡  kubectl get nodes                                                       
NAME                                        STATUS   ROLES    AGE     VERSION
ip-10-0-35-1.us-west-2.compute.internal     Ready    <none>   7m29s   v1.28.2
ip-10-0-40-151.us-west-2.compute.internal   Ready    <none>   8m46s   v1.28.3-eks-e71965b
ip-10-0-70-112.us-west-2.compute.internal   Ready    <none>   7m28s   v1.28.2
☁  eks [magzhang/update-to-k8s-1.28] ⚡  kubectl exec nvidia-driver-daemonset-8pxcm -n gpu-operator -- nvidia-smi
Tue Jan  9 19:29:40 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  Tesla V100-SXM2-16GB           On  | 00000000:00:1E.0 Off |                    0 |
| N/A   27C    P0              23W / 300W |      0MiB / 16384MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
☁  eks [magzhang/update-to-k8s-1.28] ⚡  helm ls -n gpu-operator
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
gpu-operator    gpu-operator    1               2024-01-09 14:23:37.135651533 -0500 EST deployed        gpu-operator-v23.9.1    v23.9.1  
```

## GKE 

```
~  kubectl get nodes
NAME                                            STATUS   ROLES    AGE   VERSION
gke-mz-test-tf-mz-test-cpu-pool-0f46400e-qxgn   Ready    <none>   25m   v1.28.3-gke.1203001
gke-mz-test-tf-mz-test-gpu-pool-0f4188d2-50xr   Ready    <none>   23m   v1.28.3-gke.1203001
gke-mz-test-tf-mz-test-gpu-pool-0f4188d2-n962   Ready    <none>   23m   v1.28.3-gke.1203001

~ helm ls -n gpu-operator
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
gpu-operator    gpu-operator    1               2024-01-10 16:40:36.295609783 -0500 EST deployed        gpu-operator-v23.9.1    v23.9.1 

~  kubectl exec -it pod/nvidia-driver-daemonset-c25gr -n gpu-operator -- nvidia-smi
Wed Jan 10 22:04:53 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  Tesla V100-SXM2-16GB           On  | 00000000:00:04.0 Off |                    0 |
| N/A   35C    P0              24W / 300W |      0MiB / 16384MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,9 +9,11 @@
 
 #### Coding Guidelines
 
-- All source code contributions must be formatted prior to checkin by running `terraform fmt`, and changes should not break validation checks provided by the `terraform validate` command.
+- All source code contributions must be formatted prior to checkin by running `terraform fmt -recursive`, and changes should not break validation checks provided by the `terraform validate` command.
```
Comment (ikatz-nvidia) [inline CONTRIBUTING.md:12]: Should consider adding https://github.com/marketplace/actions/terraform-validate at a future point

File: eks/terraform.tfvars [hunk 2]
```diff
@@ -35,25 +28,25 @@
...
 # nvaie                                 = false
-# nvaie_gpu_operator_driver_version     = "525.125.06"
-# nvaie_gpu_operator_version            = "v23.3.2"
+# nvaie_gpu_operator_driver_version     = "535.129.03"
+# nvaie_gpu_operator_version            = "v23.9.0"
 # private_subnets = [
-#   "10.0.1.0/24",
-#   "10.0.2.0/24",
-#   "10.0.3.0/24"
+#   "10.0.0.0/19",
```
Comment (ikatz-nvidia) [inline eks/terraform.tfvars:42]: Should probably add a comment as to why these new variables are appropriate

**Review by ikatz-nvidia — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (MaggieXJZhang) [reply CONTRIBUTING.md:12]: image
`terraform validate` is already part of this sentence. Is there a different way you want us to put it?

File: eks/terraform.tfvars (hunk 2 above)
Comment (MaggieXJZhang) [reply eks/terraform.tfvars:42]: Someone else who made a commit to change the values didn't update the default values in the tfvars file. The actual code change was https://github.com/NVIDIA/nvidia-terraform-modules/pull/17.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ikatz-nvidia) [reply CONTRIBUTING.md:12]: No, I'm saying that the github action that automatically validates pull requests should be added to this repository at a future point

Comment (MaggieXJZhang) [reply CONTRIBUTING.md:12]: I see, thanks for sharing the GitHub Actions link

**Pull request merged**